packages/dcos-integration-test/extra/test_3dt.py

```diff
@@ -556,6 +556,45 @@ def test_3dt_bundle_download_and_extract(cluster):
     # for public agent host
     expected_public_agent_files = ['dcos-mesos-slave-public.service.gz'] + expected_common_files
 
+    def _read_from_zip(z, item, to_json=True):
+        # raises KeyError if item is not in zipfile.
+        item_content = z.read(item).decode()
+
+        if to_json:
+            # raises ValueError if cannot deserialize item_content.
+            return json.loads(item_content)
+
+        return item_content
+
+    def _get_3dt_health(z, item):
+        assert isinstance(z, zipfile.ZipFile)
+        assert isinstance(item, str)
+
+        # try to load 3dt health report and validate the report is for this host
+        try:
+            health_report = _read_from_zip(z, item)
+        except KeyError as e:
+            # we did not find a key in archive, let's take a look at items in archive and try to read
+            # diagnostics logs.
+
+            # namelist() gets a list of all items in a zip archive.
+            logging.info(z.namelist())
+
+            # summaryErrorsReport.txt and summaryReport.txt are diagnostic job log files.
+            for log in ('summaryErrorsReport.txt', 'summaryReport.txt'):
+                try:
+                    log_data = _read_from_zip(z, item)
+                    logging.info("{}:\n{}".format(log, log_data))
+                except KeyError:
+                    logging.info("Could not read {}".format(log))
+            raise e
```

Comment:
since you're raising the same error, you can not catch by name, rather just use (`except KeyError:` and later `raise` at the end.